### PR TITLE
dokku_clone: add `version` parameter and switch to `git:sync`

### LINKED ISSUE #88: dokku_clone: add `version` parameter?
## Description of problem

So far, the [dokku_clone module](https://github.com/dokku/ansible-dokku#dokku_clone) supports only `app` and `repository` parameters.
Providing a path to a git repository will check out the default branch, thus making deploys not reproducible.

#### Expected Results

I didn't try whether one can pass a URL to a specific commit/tag (maybe that could also work?), but it seems to me that the usual approach in ansible is a `version` parameter (see [git module](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/git_module.html#parameter-version)).

### LINKED ISSUE #94: support new `git:sync` command
dokku 0.23 comes with a `git:sync` command https://github.com/dokku/dokku/pull/4297

This [replaces the dokku-clone](https://github.com/crisward/dokku-clone/issues/23#issuecomment-766392165) plugin; a fix for https://github.com/dokku/ansible-dokku/issues/88 is already built in.

My suggestion would be to deprecate the `dokku_clone` ansible module and replace the `dokku_git_sync` module with one that uses the dokku internals (and update the corresponding  [docs](https://github.com/dokku/ansible-dokku#dokku_git_sync) to no longer point to the commercial `git_sync` plugin).

@josegonzalez if you would like to do any of this, please go ahead.
I might have some time next weekend

### PR DESCRIPTION
**Pull request by ltalirz**
fixes #88 
fixes #94 

Use new `git:sync` command under the hood for `dokku_clone`.
Also, start reusing code between modules by placing code inside the `module_utils` folder.
Further additions for fixing tests during the `verify` stage.

Note: This requires `dokku >= 0.23.0` and will break the `dokku_clone` command for previous dokku releases.
I therefore suggest to 
- [x] first merge the other PRs that are in the pipeline
- [x] then make anew release of ansible-dokku
- [ ] then merge this PR

### CODE REVIEW

**Comment by ltalirz** [comment]
Opened https://github.com/ansible-community/molecule/issues/3045

**Comment by caillou** [comment]
Would it makes sense to expose the `--build` option of `dokku git:sync`? C.f. documentations:

> By default, this command [`dokku git:sync`] does not trigger an application build. To do so during a `git:sync`, specify the `--build` flag.

**Comment by ltalirz** [comment]
thanks for the suggestion @caillou , that would seem to make a lot of sense.

I finally got a reply on the molecule issue some time ago, just haven't found the time to get round and try fixes.
I'll try to make the time in the coming days and get this merged

**Comment by ltalirz** [comment]
P.S. Would you be interested in reviewing this PR once it's ready?

**Comment by ltalirz** [comment]
hi @caillou , I finally found the time to get back to this PR.
I've now switched to the `--build` option of `git:sync`, would you mind giving this a review to see whether it fits what you had in mind?

`dokku_clone` will remain building by default but it can now be turned off by passing `build: false`.

You don't necessarily need to go through all changes; I'm mainly concerned with `library/dokku_clone.py`

File: molecule/default/verify.yml [hunk 1]
```diff
@@ -35,3 +39,41 @@
       msg: |
         Found warnings in output of `apt update`:
         {{ apt_update.stderr }}
+
+  # Testing dokku_clone
+  - name: install rsync
+    apt:
+      name: rsync
```
Comment (ltalirz) [inline molecule/default/verify.yml:46]: hm... maybe this should be done in the role...
`git:sync` was failing when rsync is not present

**Comment by ltalirz** [comment]
@josegonzalez if you manage to have a quick look at `library/dokku_clone.py` as well that may be useful

I'll probably merge this tomorrow since it's going to be useful for the blog article

**Comment by ltalirz** [comment]
I'm going ahead and merging this; happy to include updates in follow-up PRs.

**Pull request merged**